packages/language/src/utils/uri.ts

```diff
@@ -61,4 +61,34 @@ export namespace UriUtils {
   export function normalize(uri: URI | string): string {
     return URI.parse(uri.toString()).toString();
   }
+
+  /**
+   * Compute workspace-relative parent folder for a found file candidate.
+   *
+   * Examples:
+   *  - workspace root: /repo/plugin-example
+   *  - candidate: /repo/plugin-example/cpy/test-nested-included/nested-not-here.pli
+   *  -> returns "cpy/test-nested-included"
+   *
+   * If workspace-relative cannot be computed, returns the parent folder name (e.g. "test-nested-included").
+   * Returns undefined if candidate looks invalid.
+   */
+  export function computeWorkspaceRelativeParentFolder(
+    candidateRaw: string,
+    workspaceFolderUri: URI,
+  ): string | undefined {
+    if (!candidateRaw) return undefined;
+    const candidate = normalize(candidateRaw);
```

Comment:
Suggestion: Calling normalize on the `candidateRaw` string as is doesn't work on Windows, it will result in garbled output. Use the `URI` as an input type instead.